### DISCUSSION TITLE
How should I configure cloudflared?

### DISCUSSION THREAD

**Discussion by apbarratt** [nextcloud/all-in-one#655] [Questions] [answered]
I'm really struggling to get past the landing page of AIO.  I've set up a cloudflared tunnel which is pointing at https://localhost:8080 with "No TLS Verify" turned on (without this, I could only connect to the http version and get the error message telling me to use https).

However, now that I can see the landing page on my chosen https://cloud.mydomain.tld, if I enter cloud.mydomain.tld into the New AIO Instance field, I am invariably presented with the error "Domain does not point to this server or reverse proxy not configured correctly."  This comes up even despite the fact I am looking at the server in question using the domain in question.

I see quite a few people saying they are using cloudflared with their instances so it must be possible :/

_For background, the reason I have chosen a tunnel over port forwarding is that I use a 5G Cellular Data connection and it seems IP addresses are a little more complex with such things, with the cell mast in fact holding a single IP for all phones (and routers, in my case) connected to it, effectively the mast itself is the router and I can't very well configure port forwarding on it myself.  Or so, that's the explanation I've been given for why the port forwarding built into the router does pretty much nothing.  If this is the case, I expect more and more people are going to be turning to tunnels instead of port forwarding in the coming years as cell data becomes as good (or in my case better) than a physical broadband connection._

**Comment by szaimen** [comment]
Hello, can you share your cloudflared config and did you try to follow https://github.com/nextcloud/all-in-one/blob/main/reverse-proxy.md?

**Reply by apbarratt** [reply]
Thanks for the quick response :)
I have not tried the reverse proxy stuff, no.  I'm not sure if they're the same thing as what cloudflared is doing or if it's something I should be doing in addition.  Certainly the old cloudflared config files used what it called nginx lines but I'm not sure how I'd configure that with the new stuff now that it's all handled through their zerotrust dashboard (which admitedly makes things a lot simpler to update and configure on their side!)
<img width="1315" alt="Screenshot 2022-05-18 at 19 44 05" src="https://user-images.githubusercontent.com/642320/169125376-d1dd8184-c924-4e3d-bbff-b08c6a1ccc9e.png">

**Reply by apbarratt** [reply]
p.s. I honestly don't know if it's the end of the world if I tell you my domain name or not.  I mean, it's going to be exposed to the public anyway with my trust in the user management stuff doing its job well 😅

**Comment by szaimen** [comment]
I see. Can you stop the domaincheck container with `sudo docker stop nextcloud-aio-domaincheck` and recreate the mastercontainer with 
```
# For x64 CPUs:
sudo docker run -it \
--name nextcloud-aio-mastercontainer \
--restart always \
-p 8080:8080 \
-e APACHE_PORT=11000 \
--volume nextcloud_aio_mastercontainer:/mnt/docker-aio-config \
--volume /var/run/docker.sock:/var/run/docker.sock:ro \
nextcloud/all-in-one:latest
```

Then please point your cloudflare config to `http://localhost:11000` and try to enter your domain again. Does this work?

**Reply by apbarratt** [reply]
Thanks, @szaimen.  I've given that a go but the result now is that I don't get the landing page, but a 48 character long alphanumeric string that I shan't paste here as it looks suspiciously like a cryptographic key or hash of some kind :)

**Reply by szaimen** [reply]
Yes, you must still access the aio interface locally where you need to enter the domain...

**Reply by szaimen** [reply]
> but a 48 character long alphanumeric string

That is expected! Looks good to me!

**Comment by apbarratt** [comment]
Seems I may have a new solution.  After much wrangling with my cell provider, I've been able to get a more ordinary IP address and enable port forwarding.  So I'm killing off the tunnel and setting up a reverse proxy to do the job properly.  Hopefully I can find one in a docker container so I can try and dockerise the whole thing and avoid complicated configurations on a linux box XD

**Reply by szaimen** [reply]
alright. That sounds like a better solution 👍

**Comment by szaimen** [comment, marked as answer]
Hello, I just added documentation and a flag with 1.5.0 Beta that allows to run AIO behind a Cloudflare tunnel. See https://github.com/nextcloud/all-in-one#how-to-switch-the-channel how to switch the channel and https://github.com/nextcloud/all-in-one#how-to-run-it-behind-a-cloudflare-argo-tunnel. 
Feedback is welcome!

**Comment by frtoor** [comment]


**Reply by szaimen** [reply]
Is the Aego tunnel running on the samw host and not inside a container?

**Reply by frtoor** [reply]
*Facepalm* delete this embarrassing post. I shouldn't have been up so late working. @szaimen Thank You.

**Reply by szaimen** [reply]
Great that you solved it! Cant you delete it yourself?

**Reply by frtoor** [reply]
Sure enough, right you are.
